source/MaterialXCore/Interface.cpp

```diff
@@ -319,17 +319,12 @@ InputPtr Input::getInterfaceInput() const
 
 void Input::addTokens(StringResolverPtr& resolver) const
 {
-    // If this input has an interface Input then use the tokens associated with that Input
-    // otherwise use the tokens directly associated with this Input.
-    InputPtr interfaceInput = getInterfaceInput();
-    if (interfaceInput)
-    {
-        interfaceInput->addTokens(resolver);
-    }
-    else
+    ConstNodeGraphPtr graph = getAncestorOfType<NodeGraph>();
```

Comment:
Generalize as the parent graph could have tokens which are not associated with an interface input.
Also include checking for tokens at the input level.